### ISSUE TITLE
Building an application with this client library using go mod doesn't work

### ISSUE THREAD

**Issue by vitorarins** [moby/moby#44780]
### Description

I am trying to build the example from docker website.
Unfortunately I get this error when running the `go build` command:

```sh
# github.com/docker/docker/client
vendor/github.com/docker/docker/client/errors.go:27:16: undefined: errors.As
vendor/github.com/docker/docker/client/errors.go:45:12: undefined: errors.As
vendor/github.com/docker/docker/client/request.go:137:13: undefined: errors.Is
```

Just in case it's relevant, I am mentioning that I am using `go.mod` for the module dependency.

### Reproduce

1. go mod init
2. go mod tidy
3. go mod download / go mod vendor
4. go build

### Expected behavior

`go build` should have built the example from documentation.

### docker version

```bash
Client: Docker Engine - Community
 Version:           20.10.22
 API version:       1.41
 Go version:        go1.18.9
 Git commit:        3a2c30b
 Built:             Thu Dec 15 22:28:04 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.22
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.9
  Git commit:       42c8b31
  Built:            Thu Dec 15 22:25:49 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.14
  GitCommit:        9ba4b250366a5ddde94bb7c9d1def331423aa323
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Docker Buildx (Docker Inc., v0.9.1-docker)
  scan: Docker Scan (Docker Inc., v0.23.0)

Server:
 Containers: 7
  Running: 0
  Paused: 0
  Stopped: 7
 Images: 100
 Server Version: 20.10.22
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9ba4b250366a5ddde94bb7c9d1def331423aa323
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.15.0-57-generic
 Operating System: Ubuntu 22.04.1 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.36GiB
 Name: Computer
 ID: BHJ3:PBM2:6XTK:5WJ5:I7BU:6SFS:M3SE:HS3R:HYYG:JEUE:I3C4:MYCO
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

### Additional Info

go version go1.18.6 linux/amd64

github.com/docker/docker v20.10.22+incompatible

**Comment by vvoland** [comment]
I couldn't reproduce this error on my side. Which example exactly are you trying to build?

**Comment by vitorarins** [comment]
The first one from go doc:

```
package main

import (
	"context"
	"fmt"

	"github.com/docker/docker/api/types"
	"github.com/docker/docker/client"
)

func main() {
	cli, err := client.NewClientWithOpts(client.FromEnv)
	if err != nil {
		panic(err)
	}

	containers, err := cli.ContainerList(context.Background(), types.ContainerListOptions{})
	if err != nil {
		panic(err)
	}

	for _, container := range containers {
		fmt.Printf("%s %s\n", container.ID[:10], container.Image)
	}
}
```

I will try to run it inside a `golang` container to isolate the problem.

**Comment by vitorarins** [comment]
Sorry for misleading, I actually was running the example together with a log library. Specifically this one: https://github.com/uber-go/zap
The pure example by itself works fine.

I found out that zap depends on `github.com/pkg/errors` version `0.8.1`. While the docker library depends on version `0.9.1`.
While I believe one should use the standard library `errors` package, I solved this by pinning the version to `0.9.1` and hope that doesn't break the `zap` library.

Thank you for your time.
